modules/imgproc/test/test_intelligent_scissors.cpp

```diff
@@ -147,37 +147,56 @@ void show(const Mat& img, const std::vector<Point> pts)
     }
 }
 
-TEST(Imgproc_IntelligentScissorsMB, rect)
-{
+typedef tuple<bool, size_t> IntelligentScissorsMB_TestParams;
+class Imgproc_IntelligentScissorsMB : public testing::TestWithParam<IntelligentScissorsMB_TestParams> {
+public:
     segmentation::IntelligentScissorsMB tool;
+    Imgproc_IntelligentScissorsMB()
+    {
+        const bool compute_full_local_cost = get<0>(GetParam());
+        tool.setComputeFullLocalCost(compute_full_local_cost);
+    }
+    void tool_buildMap(const Point& start_point)
+    {
+        const size_t buildMapIterations = get<1>(GetParam());
+        for (size_t idx = 1; idx <= buildMapIterations; ++idx)
+        {
+            tool.buildMap(start_point);
+        }
+    }
+};
 
+INSTANTIATE_TEST_CASE_P(/**/, Imgproc_IntelligentScissorsMB, testing::Combine(testing::Values( true, false ),
+                                                                              testing::Values( 1, 100 )));
```

Comment:
What is the purpose of adding 100 iterations for the **accuracy** test?

Reply:
It was a naive way of getting performance test numbers ... I have now in https://github.com/opencv/opencv/pull/21959/commits/cd6ef45ac2d6d820a05e9980339a5980b4195431 added a separate performance test instead.